helm/hpcc/templates/_helpers.tpl

```diff
@@ -621,37 +621,39 @@ Create placement related settings
 Pass in dict with placement
 */}}
 {{- define "hpcc.doPlacement" -}}
-{{- if .placement.nodeSelector }}
-nodeSelector:
-{{ toYaml .placement.nodeSelector | indent 2 }}
+{{- if .this.placement }}
+{{ toYaml .this.placement }}
 {{- end -}}
-{{- if .placement.tolerations }}
-tolerations:
-{{ toYaml .placement.tolerations }}
 {{- end -}}
-{{- if .placement.affinity }}
-affinity:
-{{ toYaml .placement.affinity }}
+
+{{/*
+Check if there is any placement configuration
+Pass in dict with root, pod and type
+*/}}
+{{- define "hpcc.placementsByPodType" -}}
+{{- if .root.Values.placements }}
+{{- $pod := .pod -}}
+{{- $type := printf "type:%s" .type -}}
+{{- range $placement := .root.Values.placements -}}
+{{- if or (has $pod $placement.pods) (has $type $placement.pods) -}}
+{{ include "hpcc.doPlacement" (dict "this" $placement) -}}
 {{- end -}}
-{{- if .placement.schedulerNames -}}
-{{- range $profileName := .placement.schedulerNames }}
-schedulerName: {{ $profileName }}
 {{- end -}}
 {{- end -}}
 {{- end -}}
 
 {{/*
 Check if there is any placement configuration
-Pass in dict with root, pod, cluster and type
+Pass in dict with root, pod, target and type
 */}}
-{{- define "hpcc.placementMatchName" -}}
-{{- if .root.Values.placement }}
+{{- define "hpcc.placementsByPodTargetType" -}}
```

Comment:
if this is function is exactly the same, except it also checks 'target', which I think it is,
it would be better to have one combined function, with a conditional to check if .target was set.
e.g. condition:
```
{{- if or (has $pod $placement.pods) (has $type $placement.pods) or (and .target (has $target $placement.pods)) -}}
```

Reply:
changed as suggested